components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -269,54 +270,191 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
 
         let start_key = keys::data_key(start_key);
 
-        info!(
-            self.logger,
-            "execute delete range";
-            "range_start" => log_wrappers::Value::key(&start_key),
-            "range_end" => log_wrappers::Value::key(&end_key),
-            "notify_only" => notify_only,
-            "use_delete_range" => use_delete_range,
-        );
-
+        let start = Instant::now_coarse();
         // Use delete_files_in_range to drop as many sst files as possible, this
         // is a way to reclaim disk space quickly after drop a table/index.
         if !notify_only {
-            let range = vec![EngineRange::new(&start_key, &end_key)];
-            let fail_f = |e: engine_traits::Error, strategy: DeleteStrategy| {
-                slog_panic!(
-                    self.logger,
-                    "failed to delete";
-                    "strategy" => ?strategy,
+            let (notify, wait) = oneshot::channel();
+            let delete_range = TabletTask::delete_range(
+                self.region_id(),
+                self.tablet().clone(),
+                name_to_cf(cf).unwrap(),
+                start_key.clone().into(),
+                end_key.clone().into(),
+                use_delete_range,
+                Box::new(move || {
+                    notify.send(()).unwrap();
+                }),
+            );
+            if let Err(e) = self.tablet_scheduler().schedule_force(delete_range) {
```

Comment:
should it return  error directly? Otherwise the wait.await will never finish.

Reply:
`wait.await` will return error immediately, because the `notify` is dropped on error branch.